rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -140,6 +140,10 @@ void SequentialWriter::reset()
     metadata_io_->write_metadata(base_folder_, metadata_);
   }
 
+  // if cache data exists, it must flush the data into the storage
+  if (!cache_.empty()) {
+    storage_->write(cache_);
```

Comment:
Though the method `reset` is only used in destructor now, I think we still add `cache_.clear()` after `storage_->write`.

Reply:
nice catch, i believe that is correct. w/o it, data cache and status is not consistent. I will fix them.